hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -390,6 +394,18 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	reportSilecedAlerts(hcluster)
 	reportProxyConfig(hcluster)
 
+	// Add a grace-period finalizer
+	if hcluster.Annotations == nil {
+		hcluster.Annotations = make(map[string]string)
+	}
+
+	if gracePeriod, ok := hcluster.Annotations[hyperv1.HCDestroyGracePeriodAnnotation]; ok && len(gracePeriod) > 0 && r.HCDestroyGracePeriod == 0 {
```

Comment:
you should not use a field in the reconciler (r.HCDestroyGracePeriod). A local variable is enough.